src/Database/LSMTree/Normal.hs

```diff
@@ -69,11 +82,11 @@ class SomeSerialisationConstraint a where
   Tables
 -------------------------------------------------------------------------------}
 
--- | A handle to a table.
+-- | A handle to an on-disk key\/value table.
 --
 -- An LSMT table is an individual key value mapping with in-memory and on-disk
--- parts. A table handle is the object/reference by which an in-use LSM table
--- will be operated upon. In our API it identifies a single mutable instance of
+-- parts. A table handle is the object\/reference by which an in-use LSM table
+-- will be operated upon. In this API it identifies a single mutable instance of
 -- an LSM table. The multiple-handles feature allows for there to may be many
 -- such instances in use at once.
 type TableHandle :: (Type -> Type) -> Type -> Type -> Type -> Type
```

Comment:
Addition: 

* An operation on a table handle should throw an exception if the table handle is closed.
* Closing a table handle should be idempotent IMO

Regarding sessions:
* Once a session is closed, any table handles related to that session should be closed as well.
* Using a session once it is closed, for example in `new` or `open`, should result in an exception
* Opening two sessions in the same folder should probably also result in an exception